### PR TITLE
Replace Moment.js with date-fns

### PR DESCRIPTION
**Pull request by tdonohue**
Related to #1921 

## Description
In an attempt to start to shrink the size of our main.js, this PR replaces moment.js with date-fns (which is much smaller in size)

* Why? Because Moment.js doesn't work with "tree-shaking" and it can expand the size of your bundle. See https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md

**This PR decreases the GZipped size of `main.js` by ~66KB.**  This measurement was determined by running the following on `main` vs this PR:
  * `yarn build:stats`
  * `webpack-bundle-analyzer .\dist\browser\stats.json`

## Instructions for Reviewers

* Review code
* Run some basic tests with date-based functionality to ensure no changes in behavior.
   * For example: Saving date in submission, filtering by date in search results, etc.

List of changes in this PR:
*  Created new specs for `date.util.ts` to verify the current behavior using momentjs is validated.  These tests pass with momentjs.
* Refactored code in `date.util.ts` to use `date-fns` instead, verifying the same tests still pass
* Refactored `search-range-filter.component.ts` to parse the necessary year via a new `yearFromString()` method in `date.util.ts` and added tests to verify it behaves the same as the old momentjs code.
* Removed `ngx-moment` which didn't appear to be used anywhere in our codebase.

**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes.

### CODE REVIEW

**Comment by tdonohue** [comment]
@artlowel : Assigning to you for review as you volunteered to look at shrinking main.js.  Please feel free to delegate if you wish.

File: src/app/shared/date.util.ts [hunk 1]
```diff
@@ -102,22 +88,23 @@ export function dateToNgbDateStruct(date?: Date): NgbDateStruct {
...
+  return formatInTimeZone(dateObj, 'UTC', 'yyyy-MM-dd');
 }
 
 /**
  * Checks if the given string represents a valid date
  * @param date the string to be checked
  */
 export function isValidDate(date: string) {
-  return moment(date).isValid();
+  return (isNull(date) || isUndefined(date)) ? false : isValid(new Date(date));
```
Comment (ybnd) [inline src/app/shared/date.util.ts:99]: `isNull(date) || isUndefined(date)` can be replaced with `hasNoValue(date)`

File: src/app/shared/date.util.ts [hunk 2]
```diff
@@ -31,21 +31,7 @@ export function dateToISOFormat(date: Date | NgbDateStruct | string): string {
   const dateObj: Date = (date instanceof Date) ? date :
     ((typeof date === 'string') ? ngbDateStructToDate(stringToNgbDateStruct(date)) : ngbDateStructToDate(date));
```
Comment (ybnd) [inline src/app/shared/date.util.ts:32]: Now that this method is so much simpler this nested ternary thing stands out a lot...

**Review by ybnd — COMMENTED** [review]
Confirmed that date handling still works as it did before.
The simplified `date.util.ts` is a nice bonus; I added some clean-up suggestions but I'm +1 either way

File: src/app/shared/date.util.ts (hunk 1 above)
Comment (tdonohue) [reply src/app/shared/date.util.ts:99]: Thanks @ybnd : I'll get this fixed.

File: src/app/shared/date.util.ts (hunk 2 above)
Comment (tdonohue) [reply src/app/shared/date.util.ts:32]: Agreed, @ybnd that this is a bit of ugly code. That said, I honestly couldn't find a way to easily pull this apart without requiring more refactors (which I didn't have time to work on).  So, while this is ugly, it appears to work as expected based on the new tests.  I'm planning to just leave this as-is for now, until someone has a chance/reason to clean this code up further.

Comment (ybnd) [reply src/app/shared/date.util.ts:32]: ok 👍

**Comment by tdonohue** [comment]
Merging, as [this is at +1](https://github.com/DSpace/dspace-angular/pull/1903#pullrequestreview-1158632096).  Minor feedback was addressed and tests still pass.

**Pull request merged**